CUDADataFormats/Track/interface/TrackSoAHeterogeneousT.h

```diff
@@ -19,39 +19,61 @@ class TrackSoAHeterogeneousT {
   using hindex_type = uint32_t;
   using HitContainer = cms::cuda::OneToManyAssoc<hindex_type, S, 5 * S>;
 
+
+  // quality accessors
+  constexpr Quality quality(int32_t i) const { return reinterpret_cast<Quality>(quality_(i)); }
+  constexpr Quality &quality(int32_t i) { return reinterpret_cast<Quality &>(quality_(i)); }
+  constexpr Quality const *qualityData() const { return reinterpret_cast<Quality const *>(quality_.data()); }
+  constexpr Quality *qualityData() { return reinterpret_cast<Quality *>(quality_.data()); }
+
+  // chi2 accessors
+  __host__ __device__ auto & chi2(int32_t i) { return chi2_(i); }
```

Comment:
question (I'm not sure myself): would t work to make these accessors `constexpr` ?

Reply:
also, could you make the return type explicit, instead of `auto` ?
I would like to be able to know if these return a `float` or a `double` or something else.

Reply:
I think not. Quite the opposite, all the constexpr specifiers must be ignored by the compiler except for `stride()` and nHits(), which, by the way, are the same. I will simplify further,

Reply:
well, as long as the `constexpr` is valid, having it is a simpler shorthand for `__host__ __device__`

Reply:
I missed that. This makes the file nice and homogeneous now.